app/helpers/dashboard_helper.rb

```diff
@@ -15,6 +15,8 @@ def dashboard_actions
 
   def render_widget(widget)
     render(:partial => widget.template, :locals => widget.data)
+  rescue ActionView::MissingTemplate
+    ::Foreman::Exception.new(N_("Missing template '%{template}' for widget '%{widget}'."), :widget => widget.name, :template => widget.template)
```

Comment:
Widget names should be extracted for i18n, so I think you can safely do `_(widget.name)` here to potentially get a translated version.